variables.tf

```diff
@@ -1,52 +1,95 @@
-variable "allowed_account_ids" {
-  default     = []
-  description = "Account IDs that are allowed to access the bucket/KMS key"
-  type        = list(string)
+variable "bucket_name" {
+  description = "Name of bucket to hold tf state"
+  type        = string
 }
 
-variable "bucket" {
-  default     = ""
-  description = "Name of bucket to create (do not provide if using `remote_bucket`)"
+variable "dynamo_locktable_name" {
+  default     = "tf-locktable"
+  description = "Name of lock table for terraform"
   type        = string
 }
+
 variable "kms_alias_name" {
-  default     = ""
+  default     = null
   description = "Name of KMS Alias"
   type        = string
 }
 
 variable "kms_key_id" {
-  default     = ""
-  description = "ARN for KMS key for all encryption operations."
+  default     = null
+  description = "ARN for KMS key for all encryption operations (a key will be created if this is not provided)"
   type        = string
 }
 
+variable "lifecycle_rules" {
+  description = "lifecycle rules to apply to the bucket (set to null to skip lifecycle rules)"
+
+  default = [{
+    id                            = "tfstate-expire"
+    enabled                       = true
+    expiration                    = 90
```

Comment:
Wait... does this just delete everything after 90 days?

Reply:
yes, I thought we evicted this horrible, awful default a long time ago. I've nixed it.